File: gachisa-backend/src/main/java/com/gachisa/payment/dto/PaymentCancellationResponse.java
package com.gachisa.payment.dto;

public record PaymentCancellationResponse(
        String result,
        RefundResponse refund
) {

    public static PaymentCancellationResponse cancelled() {
        return new PaymentCancellationResponse("PARTICIPATION_CANCELLED", null);
    }

    public static PaymentCancellationResponse refundRequested(RefundResponse refund) {
        return new PaymentCancellationResponse("REFUND_REQUESTED", refund);
    }
}

File: gachisa-backend/src/main/kotlin/com/gachisa/payment/dto/PaymentCancellationResponse.kt
package com.gachisa.payment.dto

data class PaymentCancellationResponse(
    val result: String,
    val refund: RefundResponse?,
) {
    companion object {
        fun cancelled(): PaymentCancellationResponse =
            PaymentCancellationResponse("PARTICIPATION_CANCELLED", null)

        fun refundRequested(refund: RefundResponse): PaymentCancellationResponse =
            PaymentCancellationResponse("REFUND_REQUESTED", refund)
    }
}
